mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksLifecycleAwareFlow.kt

```diff
@@ -83,14 +85,17 @@ private fun startedChannel(owner: Lifecycle): Channel<Boolean> {
     return channel
 }
 
-private inline fun <T : Any?> SelectBuilder<Unit>.onReceive(
+private inline fun <T : Any?, R : Any?> SelectBuilder<R>.onReceive(
     channel: ReceiveChannel<T>,
-    crossinline onClosed: () -> Unit,
-    noinline onReceive: suspend (value: T) -> Unit
+    crossinline onClosed: () -> R,
+    noinline onReceive: suspend (value: T) -> R
 ) {
     channel.onReceiveCatching {
         val result = it.getOrNull()
-        if (result === null) onClosed()
-        else onReceive(result)
+        if (result === null) {
```

Comment:
It seems null from upstream flow will be treated as closed event (in both wrapper/token impls), so it won't work as expected in case if flow emits null values. You can switch to isSuccess/isFailure/isClosed. 
```
if (it.isFailure) {
   onClosed()
} else {
   onReceive(it.getOrThrow())
}
```

Could you also please add a unit test to `MavericksLifecycleAwareFlowKtTest`, smth like this (it should fail on main/current pr):
```
    @Test
    fun testNullableFlow() = runTest(UnconfinedTestDispatcher()) {
        val flow = flowOf<Int?>(null, null)
        val owner = TestLifecycleOwner()
        val values = mutableListOf<Int?>()
        owner.lifecycle.currentState = Lifecycle.State.STARTED
        val job = flow.flowWhenStarted(owner).onEach {
            values += it
        }.launchIn(this)
        assertEquals(listOf(null, null), values)
        job.cancel()
    }
```

Reply:
@vulpeszerda let me know when you get to the above ^ and then I will merge. thanks!

Reply:
@denis-bezrukov Thanks for suggestion! 
I realized importance of test code 😓  my bad. 

BTW, how about checking result with `isClosed` instead of `isFailure`? 
Of cource It is not so meaningful in this code, because source channel will not fail in all case (if source channel fail, exception will be thrown in `produce { collect { send(it) } }`) ,but I think handling failure case in `getOrThrow` phrase is much readable.

Reply:
@denis-bezrukov And your test code works great! 👍

Reply:
@vulpeszerda  Yep, since failure isn't possible I agree it's better to just use isClosed

Reply:
@denis-bezrukov @elihart I applied above suggestions